### PR TITLE
Write mdx documentation for Icon component

### PR DESCRIPTION
**Pull request by inhibitor1217**
## Self Checklist

- [x] I wrote a PR title in **English**.
- [x] I added an appropriate **label** to the PR.
- [x] I wrote a commit message in **English**.
- [x] I wrote a commit message according to [**the Conventional Commits specification**](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] [I added the appropriate **changeset**](https://github.com/channel-io/bezier-react/blob/next-v1/CONTRIBUTING.md#add-a-changeset) for the changes.
- [ ] I wrote **a unit test** about the implementation.
- [x] I wrote **a storybook document** about the implementation.
- [ ] I tested the implementation in **various browsers**.
  - Windows: Chrome, Edge, (Optional) Firefox
  - macOS: Chrome, Edge, Safari, (Optional) Firefox

## Summary

`Icon` 컴포넌트에 대한 Storybook MDX documentation을 작성합니다.

## Details

생각해보니.. chromatic build가 있어서 스크린샷을 올려볼 필요가 없군요 😁 

스크린샷 2022-07-28 오후 5 38 35
스크린샷 2022-07-28 오후 5 38 43
스크린샷 2022-07-28 오후 5 38 50
스크린샷 2022-07-28 오후 5 38 57
스크린샷 2022-07-28 오후 5 39 08

## Breaking change or not (Yes/No)

None

## References

None

### CODE REVIEW

**Review by aooen — APPROVED** [review]

**Review by yangwooseong — APPROVED** [review]
👍

File: packages/bezier-react/src/components/Icon/Icon.mdx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+```tsx
+<Button
+  leftContent="plus"
+  text="Add content"
+  rightContent="arrow-right"
+/>
+```
+
+> 💡 현재 composite component가 아이콘을 사용하는 방식으로는 tree-shaking 효과를 얻기가 어렵습니다.
+> Icon name을 prop으로 전달하는 방법은 다른 패턴으로 개선되어야 합니다.
```
Comment (sungik-choi) [inline packages/bezier-react/src/components/Icon/Icon.mdx:101]: 👍

File: packages/bezier-react/src/components/Icon/Icon.mdx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+### How to add new icons to `bezier-react`
```
Comment (sungik-choi) [inline packages/bezier-react/src/components/Icon/Icon.mdx:103]: 👍

추후 아이콘 빌드 과정도 자동화할 수 있도록 챙겨볼 예정입니다.

File: packages/bezier-react/src/components/Icon/Icon.mdx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+디자이너가 새로운 아이콘을 추가하거나, 기존의 아이콘을 변경한 경우 아이콘 리소스를 변경하여 배포가 이루어져야 합니다.
+
+아이콘 리소스는 SVG 파일로 이루어져 있습니다.
+이를 design tool 에서 직접 수동으로 다운로드 받을 수도 있지만, 이 프로세스를 자동화하기 위한 tool이 준비되어 있습니다.
+
+아래 문서를 참고 바랍니다.
+- [Integration with Figma](https://github.com/channel-io/bezier-react/tree/next-v1/packages/bezier-figma-plugin)
+
+### Usage (Legacy)
```
Comment (sungik-choi) [inline packages/bezier-react/src/components/Icon/Icon.mdx:113]: 👍

**Review by sungik-choi — APPROVED** [review]

**Pull request merged**